cmd/attach/policy/attach_policy_service.go

```diff
@@ -0,0 +1,153 @@
+		AWSClient: AWSClient,
+	}
+}
+
+// ValidateOptions validate the options:
+// verify rolename and each policy arn are valid
+// verify role is RedHat managed
+// verify role has quota to attach the policies
+func (p *policyService) ValidateOptions(roleName string, policyArns []string) error {
+	role, err := p.AWSClient.GetRoleByName(roleName)
+	if err != nil {
+		return fmt.Errorf(
+			"Failed to find the role %s: %s",
+			roleName, err,
+		)
+	}
+	isRedHatManaged := false
+	for _, tag := range role.Tags {
+		if awsutil.ToString(tag.Key) == tags.RedHatManaged &&
+			awsutil.ToString(tag.Value) == "true" {
+			isRedHatManaged = true
+			break
+		}
+	}
+	if !isRedHatManaged {
+		return fmt.Errorf("Rosa only supports attaching policy to RedHat managed role")
+	}
+
+	err = validatePolicyQuota(p.AWSClient, roleName, policyArns)
+	if err != nil {
+		return err
+	}
+	for _, policyArn := range policyArns {
+		fmt.Printf("validate policy %s\n", policyArn)
+		_, err = p.AWSClient.IsPolicyExists(policyArn)
+		if err != nil {
+			return fmt.Errorf(
+				"Failed to find the policy %s: %s",
+				policyArn, err,
+			)
+		}
+	}
+	return nil
+}
+
+func (p *policyService) AutoAttachArbitraryPolicy(roleName string, policyArns []string,
+	accountID, orgID string) (string, error) {
+	output := ""
+	for _, policyArn := range policyArns {
+		err := p.AWSClient.AttachRolePolicy(roleName, policyArn)
+		if err != nil {
+			return output, fmt.Errorf("Failed to attach policy %s to role %s", policyArn, roleName)
+		}
+		output = output + fmt.Sprintf("Attached policy '%s' to role '%s'\n", policyArn, roleName)
+		p.OCMClient.LogEvent("ROSAAttachPolicyAuto", map[string]string{
+			ocm.Account:      accountID,
+			ocm.Organization: orgID,
+			ocm.RoleName:     roleName,
+			ocm.PolicyArn:    policyArn,
+		})
+	}
+	return output, nil
+}
+
+func (p *policyService) ManualAttachArbitraryPolicy(roleName string, policyArns []string,
+	accountID, orgID string) string {
+	cmd := ""
+	for _, policyArn := range policyArns {
+		cmd = cmd + fmt.Sprintf("aws iam attach-role-policy --role-name %s --policy-arn %s\n",
+			roleName, policyArn)
+		p.OCMClient.LogEvent("ROSAAttachPolicyManual", map[string]string{
+			ocm.Account:      accountID,
+			ocm.Organization: orgID,
+			ocm.RoleName:     roleName,
+			ocm.PolicyArn:    policyArn,
+		})
+	}
+	return cmd
+}
+
+func validatePolicyQuota(c aws.Client, roleName string, policyArns []string) error {
+	quota, err := c.GetIAMServiceQuota(quotaCode)
+	if err != nil {
+		return fmt.Errorf("Failed to get quota of policies per role: %s", err)
+	}
+	attachedPolicies, err := c.GetAttachedPolicy(&roleName)
+	if err != nil {
+		return fmt.Errorf("Failed to get attached policies of role %s: %s", roleName, err)
+	}
+	if len(policyArns)+len(attachedPolicies) > int(*quota.Quota.Value) {
+		policySkipped := 0
+		for _, attachedPolicy := range attachedPolicies {
+			if slices.Contains(policyArns, attachedPolicy.PolicyArn) {
+				policySkipped++
```

Comment:
Would it be better to output a message back to the user, the list of already attached policy arns?

Reply:
@davidleerh 
seems user is not always interested in such info. The message including all attached policies will make the output redundancy.
Maybe we can just output such info only if the quota is exceeded.
wdyt?

Reply:
IMHO getting the output to show already included policy arns when the quota is exceeded seems redundant as well. I am fine to leave it as it. Self resolving the comment. Thanks for looking.